tests/blobcache.bats

```diff
@@ -55,6 +55,36 @@ function _check_matches() {
         expect_output --from="$unmatched" "$5"  "$6 should not match"
 }
 
+# Integration test for https://github.com/containers/image/pull/1645
+@test "blobcache: blobs must be reused when pushing across registry" {
+	start_registry
+	run_buildah login --tls-verify=false --authfile ${TEST_SCRATCH_DIR}/test.auth --username testuser --password testpassword localhost:${REGISTRY_PORT}
+	run_buildah pull registry.fedoraproject.org/fedora-minimal
+	run_buildah tag registry.fedoraproject.org/fedora-minimal localhost:${REGISTRY_PORT}/test
+	run_buildah --log-level debug push --tls-verify=false --authfile ${TEST_SCRATCH_DIR}/test.auth localhost:${REGISTRY_PORT}/test docker://localhost:${REGISTRY_PORT}/test
```

Comment:
Is this actually correct?

This push could create a _different_ compressed configuration than the original one at `r.fp.org`. And if so, after we clear the BIC, I don’t know how it would be matched on the second push.

The “How to manually test this” paragraph uses `skopeo copy` to create a possibly-matched copy on the second registry, and IIRC that’s necessary.

Reply:
@mtrmac After clearing the BIC, we pull again which populates the BIC with candidate with a specific case which is candidate with blob and `UnknownLocation: True`.

I think clearing BIC and image from local produces similar effect to `skopeo copy` , doesn't it ? and further this can be verified by running the same test with older c/image where feature is not present, there test fails.

Given that' I agree `skopeo copy` is more clean and I'll amend this patch to use `skoepo copy` instead.

Reply:
The variable here is whether `buildah push` produces a bit-for-bit-identical layer to the original `r.fp.org` one. That might be _consistently_ true today, but it will very likely eventually not be true (after the implementations and/or implementation versions between the Buildah being tested, and the thing that uploaded to `r.fp.org`, drift).

With an empty BIC, pulling records (uncompressed, compressed as in `r.fp.org`, location = r.fp.org) in BIC.
Then, the push triggers a search for (compressed as in `r.fp.org`, unknown location); that matches only if the push to `localhost` produces a bit-for-bit identical layer.

I guess that it does right now, but that can change any time.

Reply:
@mtrmac PTAL again, not used `skopeo copy` but now image is built locally and pulled from a dir, where this should never happen. I have tested this test with older buildah ( where this fails ) and newer buildah ( with vendor ) where this passes.